analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/TypeTypeFilter.java

```diff
@@ -2,13 +2,27 @@
 
 import java.util.Objects;
 
+/**
+ * I don't know exactly what this is for ?.
+ */
 public class TypeTypeFilter {
 
+  /**
+   * Check if given object is a TypeType.CLASS ?.
+   *
+   * @param obj ?
+   * @return ?
```

Comment:
This checks if the obj is of type class, if so then that object should be able to be serialized.

Reply:
Other way around; to follow what the C# implementation does, we do not serialize `"Type": 0` but only serialize the values that are not the default.